### PR TITLE
Preview of Pymer4 PR

### PR DESCRIPTION
**Pull request by AndrewRidden-Harper**
This PR is a preview of what the Pymer4 developers will see. I modified Pymer4 so that the full output of the R ranef function is stored as a Pandas DataFrame (using the R as.data.frame method). We only needed the conditional standard deviation (condsd) column as the information in the other columns was already available in Pymer4. However, getting the whole ranef dataframe was the easiest method. I also added a test and a documentation example for the new feature. Following the contribution guidelines, I made the branch off their dev branch.

### CODE REVIEW

**Comment by AndrewRidden-Harper** [comment]
Getting the whole ranef dataframe also means that there is no risk of accidentally associating conditional standard deviations with the wrong random effect as site and event names as are also included in the dataframe

File: pymer4/models/Lmer.py [hunk 1]
```diff
@@ -5,28 +5,29 @@
...
+from copy import copy
+
+import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
-import matplotlib.pyplot as plt
+import rpy2.rinterface as rinterface
+import rpy2.robjects as robjects
 import seaborn as sns
+from pandas.api.types import CategoricalDtype
```
Comment (sungeunbae) [inline pymer4/models/Lmer.py]: Is this order of import ok to your PyCharm or isort (I'm curious!)

File: examples/example_01_basic_usage.py [hunk 2]
```diff
@@ -115,6 +115,22 @@
 # Get group level deviates from population level coefficients (i.e. rfx)
 print(model.ranef.head(5))
 
+###############################################################################
+
+# Get group level deviates from population level coefficients (also called conditional modes) as well as their
+# associated conditional standard deviations in the form of a dataframe
+
+# Get a dataframe containing information about the random effects. The dataframe contains the following columns:
```
Comment (lispandfound) [inline examples/example_01_basic_usage.py]: Is this dataframe `model.ranef_df`? If so, maybe mention it.
```suggestion
# :code:`model.ranef_df` is a dataframe containing information about the random effects. The dataframe contains the following columns:
```

**Review by lispandfound — CHANGES_REQUESTED** [review]

**Review by claudio525 — APPROVED** [review]

File: pymer4/models/Lmer.py (hunk 1 above)
Comment (AndrewRidden-Harper) [reply pymer4/models/Lmer.py]: isort actually was ok with most of it.

It only changed these orders:

from ..bridge import R2numpy, R2pandas, con2R, pandas2R
from ..utils import (
    _perm_find,
    _return_t,
    _sig_stars,
    _to_ranks_by_group,
)

**Pull request closed without merging**